app/gui2/src/util/ast/extended.ts

```diff
@@ -19,6 +19,12 @@ import {
 } from '.'
 import type { Opt } from '../opt'
 
+type ExtractType<V, T> = T extends ReadonlyArray<infer Ts>
+  ? Extract<V, { type: Ts }>
+  : Extract<V, { type: T }>
+
+type OneOrArray<T> = T | readonly [...T[]]
```

Comment:
might want to add a comment noting that the spread here is to force inference to infer a tuple? although i don't think anything needs the type to be a tuple, right?

Reply:
unless it's required for correct inference, i suppose

Reply:
Seems like it is not actually necessary. `T[]` works just fine. I probably had an inference problem somewhere else when I introduced that :)